system_metrics_collector/src/system_metrics_collector/collector.hpp

```diff
@@ -28,26 +31,45 @@ class Collector
   virtual ~Collector() = default;
 
   /**
-   * Start collecting data. Meant to be called after construction.
+   * Start collecting data. Meant to be called after construction. Note: this locks the recursive mutex class
+   * member 'mutex'.
+   *
+   * @return true if started, false if an error occurred
+   */
+  bool start();
+
+  /**
+   * Override in order to perform necessary starting steps. Overridden methods should lock the recursive mutex class
+   * member 'mutex'
    *
-   * @return true if started, false if an error occured
+   * @return true if setup was successful, false otherwise.
    */
-  virtual bool start() = 0;
+  virtual bool setupStart() = 0;
```

Comment:
- Could we move this protected?
- Let's rename this to `setupStartLocked` and document the mutex need to be locked before the function is being called

Reply:
Thanks for the articles above! 

I think the setup methods should be moved to private. Doing so allows us to use a single mutex in the collector class and removing the locking behavior in the periodic_measurement_node class.